core/src/main/java/org/apache/brooklyn/core/objs/proxy/InternalEntityFactory.java

```diff
@@ -46,46 +48,60 @@
 import org.apache.brooklyn.util.core.flags.FlagUtils;
 import org.apache.brooklyn.util.core.task.Tasks;
 import org.apache.brooklyn.util.exceptions.Exceptions;
+import org.apache.brooklyn.util.guava.Maybe;
 import org.apache.brooklyn.util.javalang.AggregateClassLoader;
 import org.apache.brooklyn.util.javalang.Reflections;
 import org.apache.brooklyn.util.text.Strings;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.annotation.Nonnull;
 import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.Objects;
 import java.util.Queue;
 import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 import static com.google.common.base.Preconditions.checkState;
 
 /**
- * Creates entities (and proxies) of required types, given the 
- * 
+ * Creates entities (and proxies) of required types, given the
+ *
  * This is an internal class for use by core-brooklyn. End-users are strongly discouraged from
  * using this class directly.
- * 
+ *
  * Used in three situations:
  * <ul>
  *   <li>Normal entity creation (through entityManager.createEntity)
  *   <li>rebind (i.e. Brooklyn restart, or promotion of HA standby manager node)
  *   <li>yaml parsing
  * </ul>
- * 
+ *
  * @author aled
  */
 public class InternalEntityFactory extends InternalFactory {
 
     private static final Logger log = LoggerFactory.getLogger(InternalEntityFactory.class);
-    
+
     private final EntityTypeRegistry entityTypeRegistry;
     private final InternalPolicyFactory policyFactory;
     private final ClassLoaderCache classLoaderCache;
-    
+
+    /**
+     * The initializers to be add to any application deployed by Brooklyn.
+     * e.g. <code>brooklyn.deployment.initializers=org.apache.brooklyn.core.effector.AddDeploySensorsInitializer</code>
+     * will automatically add tags to the root node of any deployed application.
+     */
+    public final static ConfigKey<String> DEFAULT_INITIALIZERS_CLASSNAMES = ConfigKeys.newStringConfigKey(
```

Comment:
`DEFAULT` feels wrong -- should we call this `GLOBAL_DEPLOYMENT_INITIALIZERS` ?

(and change the method `getDefaultInitializers()`)

Reply:
@ahgittin Should I also change the property name or are you happy with what is there already?

Reply:
`brooklyn.deployment.initializers` is a great name - i should have said!